lib/logstash/inputs/syslog.rb

```diff
@@ -58,8 +61,10 @@ def register
       "match" => { "message" => "<%{POSINT:priority}>%{SYSLOGLINE}" },
     )
 
+    locale = @config["locale"][0] if @config["locale"] != nil and @config["locale"][0] != nil
```

Comment:
Hi @nwknu, we should remove this check as @colinsurprenant pointed out and directly pass it to the date filter if it is not `nil` or empty